src/container/tabs.rs

```diff
@@ -0,0 +1,178 @@
+use egui::{vec2, Rect};
+
+use crate::{
+    is_being_dragged, Behavior, DropContext, InsertionPoint, LayoutInsertion, TileId, Tiles, Tree,
+};
+
+/// A container with tabs. Only one tab is open (active) at a time.
+#[derive(Clone, Debug, Default, serde::Serialize, serde::Deserialize)]
+pub struct Tabs {
+    /// The tabs, in order.
+    pub children: Vec<TileId>,
+
+    /// The currently open tab.
+    pub active: TileId,
+}
+
+impl Tabs {
+    pub fn new(children: Vec<TileId>) -> Self {
+        let active = children.first().copied().unwrap_or_default();
+        Self { children, active }
+    }
+
+    pub fn add_child(&mut self, child: TileId) {
+        self.children.push(child);
+    }
+
+    pub fn set_active(&mut self, child: TileId) {
+        self.active = child;
+    }
+
+    pub(super) fn layout<Pane>(
+        &mut self,
+        tiles: &mut Tiles<Pane>,
+        style: &egui::Style,
+        behavior: &mut dyn Behavior<Pane>,
+        rect: Rect,
+    ) {
+        if !self.children.iter().any(|&child| child == self.active) {
+            // Make sure something is active:
+            self.active = self.children.first().copied().unwrap_or_default();
+        }
+
+        let mut active_rect = rect;
+        active_rect.min.y += behavior.tab_bar_height(style);
+
+        // Only lay out the active tab (saves CPU):
+        tiles.layout_tile(style, behavior, active_rect, self.active);
+    }
+
+    pub(super) fn ui<Pane>(
+        &mut self,
+        tree: &mut Tree<Pane>,
+        behavior: &mut dyn Behavior<Pane>,
+        drop_context: &mut DropContext,
+        ui: &mut egui::Ui,
+        rect: Rect,
+        tile_id: TileId,
+    ) {
+        let next_active = self.tab_bar_ui(tree, behavior, ui, rect, drop_context, tile_id);
+
+        // When dragged, don't show it (it is "being held")
+        let is_active_being_dragged = is_being_dragged(ui.ctx(), self.active);
+        if !is_active_being_dragged {
+            tree.tile_ui(behavior, drop_context, ui, self.active);
+        }
```

Comment:
that's an interesting difference to egui_dock and I think worse here? In egui dock one only drags the tab title always and the view is still visible in that time instead of leaving a black hole